backend/query/test/items.e2e-spec.ts

```diff
@@ -1,42 +1,185 @@
+        initializeMockModule(),
+        ItemsModule,
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalPipes(
+      new ValidationPipe({
+        whitelist: true,
+        transform: true,
+        transformOptions: { enableImplicitConversion: true },
+      }),
+    );
+
+    itemModel = moduleFixture.get('ItemModel');
+    tagModel = moduleFixture.get('TagModel');
+    itemsByTagModel = moduleFixture.get('ItemsByTagModel');
+
+    app.setGlobalPrefix('queries/v1');
+    await app.init();
+    server = app.getHttpServer();
   });
 
   beforeEach(async () => {
-    // await itemRepository.clear();
-    // await itemRepository.insert(singleItem);
+    await itemModel.deleteMany();
+    await tagModel.deleteMany();
+    await itemsByTagModel.deleteMany();
+
+    await itemModel.insertMany(items);
+    await tagModel.insertMany(tags);
+    await itemsByTagModel.insertMany(itemsByTags);
   });
 
   afterAll(async () => {
-    // await itemRepository.clear();
-    //await teardownMockDataSource();
-    //await app.close();
+    await itemModel.deleteMany();
+    await tagModel.deleteMany();
+    await itemsByTagModel.deleteMany();
+    await teardownMockDataSource();
+    server.close();
+    await app.close();
   });
 
-  describe('Queries /query/v1', () => {
-    // const queriesPath = '/query/v1/';
+  describe('Queries /queries/v1', () => {
+    const queriesPath = '/queries/v1/';
 
-    it('/ => mock', async () => {
-      expect(true).toBe(true);
-    });
+    describe('items/list', () => {
+      const subPath = 'items/list';
+
+      it('should return 16 items searching for android (74 total)', async () => {
+        const result = await request(server)
+          .get(queriesPath + subPath)
+          .query({ query: 'android' })
+          .expect(HttpStatus.OK);
+
+        expect(result.body).toHaveLength(16);
+      });
+
+      it('should return 16 items searching with tags for android (74 total)', async () => {
+        const result = await request(server)
+          .get(queriesPath + subPath)
+          .query({ tags: ['android'] })
+          .expect(HttpStatus.OK);
+
+        expect(result.body).toHaveLength(16);
+      });
+
+      it('should not return items if we remove the tags containing android', async () => {
+        await itemModel.updateMany({}, { $set: { tags: [] } });
+
+        const result = await request(server)
+          .get(queriesPath + subPath)
+          .query({ query: 'android' })
+          .expect(HttpStatus.OK);
 
-    // it('/:slug => getOneItem', async () => {
-    //   const res = await request(app.getHttpServer())
-    //     .get(queriesPath + 'get-one-item/' + singleItem.slug)
-    //     .expect(HttpStatus.OK);
-    //   expect(res.body.name).toBe(singleItem.name);
-    // });
+        expect(result.body).toHaveLength(0);
+      });
+
+      it('should return first the items that match better', async () => {
```

Comment:
This description is weird => "Should consistently sort by relevance" would be better